mainapp/utils/model_validator.py

```diff
@@ -0,0 +1,73 @@
+import json
+import logging
+import os
+import subprocess
+import tempfile
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+def validate_glb_file(file_field):
+    """
+    Validates a GLB file using the Khronos Group's gltf-validator CLI.
+    Raises:
+        ValidationError: if the file is not a valid GLB or fails validation.
+    """
+
+    if not file_field.name.lower().endswith(".glb"):
+        raise ValidationError("Only .glb files are supported.")
+
+    file_field.seek(0)
+    header = file_field.read(4)
+    if header != b"glTF":
+        raise ValidationError(
+            "The uploaded file does not appear to be a valid GLB file."
+        )
+
+    try:
+        file_field.seek(0)
+        with tempfile.NamedTemporaryFile(suffix=".glb") as temp_file:
+            for chunk in file_field.chunks():
+                temp_file.write(chunk)
+            temp_file.flush()
+
+            result = subprocess.run(
+                [settings.GLTF_VALIDATOR, temp_file.name, "-o"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            output = json.loads(result.stdout.decode("utf-8"))
+
+            # Ensures the model has some shape. At least a cube.
+            if output["info"]["totalVertexCount"] < 8 or \
```

Comment:
If the fields are not present, this will throw a KeyError and end in an internal server error, which may or may not be what we want. How stable is the result format here? Are the fields supposed to be present unless the validator changes the format in a new version? Can we be sure the info fields are present, when there was an error in the model, i.e. may be check for errors first?